bravado/mapping/formatter.py

```diff
@@ -1,66 +1,83 @@
 
     :param spec: spec for a primitive type as a dict
     :type value: int, long, float, boolean, string, unicode, etc
-    :return: value (default may be provided in the spec)
-    :raises: SwaggerError in validation failure
+    :rtype: int, long, float, boolean, string, unicode, etc
     """
-    if not schema.has_format(spec):
+    if value is None or not schema.has_format(spec):
         return value
+
     to_wire, _, _ = _formatters[schema.get_format(spec)]
     return to_wire(value)
 
 
-def from_wire(spec, value):
-    """Unformats a value if its spec has a 'format' specifier.
+def to_python(spec, value):
+    """Converts a value in wire format to its python representation given
+    the 'format' in the given spec.
 
     :param spec: spec for a primitive type as a dict
     :type value: int, long, float, boolean, string, unicode, etc
-    :return: value (default may be provided in the spec)
-    :raises: SwaggerError in validation failure
+    :rtype: int, long, float, boolean, string, object, etc
     """
-    if not schema.has_format(spec):
+    if value is None or not schema.has_format(spec):
         return value
-    _, from_wire, _ = _formatters[schema.get_format(spec)]
-    return from_wire(value)
-
-
-def date_to_wire(value):
-    # TODO: None handling
-    if type(value) == datetime.date:
-        # TODO: Fix me - this is obviously not correct
-        return str(value)
 
+    _, to_python, _ = _formatters[schema.get_format(spec)]
+    return to_python(value)
 
-def date_from_wire(value):
-    # TODO: None handling
-    # TODO: Fix me - this is obviously not correct
-    value = datetime.date(value)
-    return value
 
-# TODO: add other 'format' types
-# TODO: revisit int64 formatters
+# Default registered formats
 _formatters = {
-    'date': (date_to_wire, date_from_wire, 'Converts a date to/from a string'),
-    'int64': (lambda i: long(i), lambda i: long(i),
-              'Converts an integer to/from int64')
+    'date': (
+        lambda d: d.isoformat(),
+        lambda d: dateutil.parser.parse(d).date(),
+        'Converts string:date <=> python datetime.date'),
+    'date-time': (
+        lambda dt: dt.isoformat(),
+        lambda dt: dateutil.parser.parse(dt),
+        'Converts string:date-time <=> python datetime.datetime'),
+    'int64': (
+        lambda i: i if isinstance(i, long) else long(i),
+        lambda i: i if isinstance(i, long) else long(i),
+        'Converts integer:int64 <==> python long'),
+    'int32': (
+        lambda i: i if isinstance(i, int) else int(i),
+        lambda i: i if isinstance(i, int) else int(i),
+        'Converts integer:int32 <==> python int'),
+    'float': (
+        lambda f: f if isinstance(f, float) else float(f),
+        lambda f: f if isinstance(f, float) else float(f),
+        'Converts number:float <==> python float'),
+    'double': (
+        lambda d: d if isinstance(d, float) else float(d),
+        lambda d: d if isinstance(d, float) else float(d),
```

Comment:
Switching it later is difficult because it's a non-backwards compatible change. We'd have to make a whole major version bump to get that in.

Since these are really just conventions, we could do float -> float, double -> Decimal ?